src/main/java/com/minecolonies/coremod/colony/jobs/AbstractJob.java

```diff
@@ -161,12 +165,12 @@ public void readFromNBT(@NotNull final NBTTagCompound compound)
      *
      * @param tasks EntityAITasks list to add tasks to.
      */
-    public void addTasks(@NotNull final EntityAITasks tasks)
+    public void addWorkerAIToTaskList(@NotNull final EntityAITasks tasks)
     {
-        final AbstractAISkeleton<? extends AbstractJob> aiTask = generateAI();
-        if (aiTask != null)
+        workerAI = generateAI();
```

Comment:
we could set this in a "getOrGenerate" method?

Reply:
ye that should work aswell